### ISSUE TITLE
[3.3.5] Core/Creature: Creatures not evading properly if affected by damage over time effects

### ISSUE THREAD

**Issue by Treeston** [TrinityCore/TrinityCore#16461]
3.3.5 d5868a7db31a13ac147a66a25cb4c3a6b8a93f8a

Repro case:
* Take any character that can use damage over time effects (I used a cat druid)
* Engage a creature (I used Tuskarr in Nagrand, no idea if it works for others)
* Place a damage over time effect on the creature, then die

Expected behavior:
* The creature immediately realizes its threat list is empty and enters evade mode, removing the debuff and returning to spawn

Observed behavior:
* The creature fails to enter evade mode until a few seconds _after_ the final damage over time effect has expired. It will display an empty threat list (.debug threat), but will still be in combat (UNIT_STATE_COMBAT in .npc info) and will not regenerate health
* If all players leave the creature's grid (release spirit) while it is in this state, any debuffs present on the creature will become "stuck", never expiring and instead remaining perpetually with a 0 second duration (unless something causes the debuff to be removed)

**Comment by Demonid** [comment]
Confirmed on Halion

406584839c9ea7617f67e286d6d2b0fa93a7d499

**Comment by tkrokli** [comment]
While testing a level 57 death knight character in New Hearthglen (if I do not use any gm cheat tricks to survive), after aggroing a group of humans from spreading fear in a group, I find that the last target I engaged before I die will stay focused on my corpse for a while after I died. The other surviving mobs evade immediately (I have not checked yet if the diseases successfully spread to the other enemy targets). This scenario should be fairly easy to reproduce in the phases where there are still some groups of unarmed civilians left in New Hearthglen. I have not noticed it as often with single targets, then again I have not thought of conditions to check for when this happens). Last time I noticed this: 2 weeks ago while testing, TrinityCore rev. c606ac5 2016-01-20 01:38:30 +0100 (3.3.5 branch) (Win64, Release),
